src/autopas/utils/DoubleSet.h

```diff
@@ -0,0 +1,207 @@
+  /**
+   * Create a DoubleSet from a std::set
+   * @param values
+   */
+  DoubleFiniteSet(std::set<double> values) : _set(values) {}
+
+  std::unique_ptr<DoubleSet> clone() const override { return std::make_unique<DoubleFiniteSet>(*this); }
+
+  operator std::string() const override {
+    if (_set.empty()) {
+      return "{ }";
+    }
+
+    auto it = _set.begin();
+    std::stringstream ss;
+    ss << "{" << *it;
+    for (++it; it != _set.end(); ++it) {
+      ss << ", " << *it;
+    }
+    ss << "}";
+
+    return ss.str();
+  }
+
+  bool isFinite() const override { return true; }
+
+  double getMin() const override { return *_set.begin(); }
+  double getMax() const override { return *_set.rbegin(); }
+
+  std::set<double> getAll() const override { return _set; }
+
+  std::vector<double> uniformSample(unsigned n, std::default_random_engine& rng) const override {
+    std::vector<double> result;
+    result.reserve(n + _set.size());
+
+    // copy the whole set until result is full
+    while (result.size() < n) {
+      for (auto val : _set) {
+        result.push_back(val);
+      }
+    }
+
+    // if too many elements added
+    if (result.size() > n) {
+      // randomize the last copy of the set
+      size_t extra = result.size() - n;
+      std::shuffle(std::end(result) - extra, std::end(result), rng);
+
+      // truncate the rest
+      result.resize(n);
+    }
+
+    // randomize the sample
+    std::shuffle(std::begin(result), std::end(result), rng);
+
+    return result;
+  }
+};
+
+/**
+ * Class describing a interval
+ */
+class DoubleInterval : public DoubleSet {
+ private:
+  double _min;
+  double _max;
+
+ public:
+  /**
+   * Default Constructor: Create a range which only contains 0
+   */
+  DoubleInterval() : _min(0.), _max(0.) {}
+  /**
+   * Create a range which only contains given value
+   * @param val
+   */
+  DoubleInterval(double val) : _min(val), _max(val) {}
+  /**
+   * Create a range with given bounds
+   * @param min Lower bound of range.
+   * @param max Upper bound of range.
+   */
+  DoubleInterval(double min, double max) : _min(min), _max(max) { assert(min <= max); }
+
+  std::unique_ptr<DoubleSet> clone() const override { return std::make_unique<DoubleInterval>(*this); }
+
+  operator std::string() const override {
+    std::stringstream ss;
+    ss << "[" << _min << ", " << _max << "]";
+    return ss.str();
+  }
+
+  double getMin() const override { return _min; }
+  double getMax() const override { return _max; }
+
+  bool isFinite() const override { return _max == _min; }
+
+  std::set<double> getAll() const override {
+    if (isFinite()) return {_min};
+
+    utils::ExceptionHandler::exception("DoubleInterval: Interval is not finite [{}, {}]", _min, _max);
+    return {};
+  }
+
+  std::vector<double> uniformSample(unsigned n, std::default_random_engine& rng) const override {
+    std::vector<double> result;
+    result.reserve(n);
```

Comment:
```suggestion
    if(n == 0) {
      return result;
    }
    result.reserve(n);
```
Otherwise, an integer overflow will occur for `n==0` and the resulting sample will have a size of (2^32) - 1